taxi/views.py

```diff
@@ -17,3 +18,35 @@ def index(request):
     }
 
     return render(request, "taxi/index.html", context=context)
+
+
+class ManufacturerListView(generic.ListView):
+    model = Manufacturer
+    context_object_name = "manufacturer_list"
+    template_name = "taxi/manufacturer_list.html"
+    paginate_by = 5
+
+
+class CarListView(generic.ListView):
+    model = Car
+    paginate_by = 5
+    queryset = Car.objects.all().select_related("manufacturer")
+
+
+class CarDetailView(generic.DetailView):
+    model = Car
+
+
+class DriverListView(generic.ListView):
+    model = Driver
+    paginate_by = 5
+
+
+class DriverDetailView(generic.DetailView):
+    model = Driver
+
+    def get_queryset(self, **kwargs):
+        query_set = Driver.objects.filter(id=self.kwargs["pk"])
+        return query_set.prefetch_related(
+            "cars__drivers", "cars__manufacturer"
+        )
```

Comment:
FYI
No need to create this function
Try:
```suggestion
    queryset = Driver.objects.prefetch_related(
            "cars__drivers", "cars__manufacturer"
        )
```